torch/utils/tensorboard/writer.py

```diff
@@ -0,0 +1,714 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+import six
+import time
+
+from tensorboard.compat.proto.event_pb2 import SessionLog
+from tensorboard.compat.proto.event_pb2 import Event
+from tensorboard.compat.proto import event_pb2
+from tensorboard.summary.writer.event_file_writer import EventFileWriter
+
+from ._convert_np import make_np
+from ._embedding import make_mat, make_sprite, make_tsv, append_pbtxt
+from ._onnx_graph import load_onnx_graph
+from ._pytorch_graph import graph
+from ._utils import figure_to_image
+from .summary import (
+    scalar, histogram, histogram_raw, image, audio, text,
+    pr_curve, pr_curve_raw, video, custom_scalars, image_boxes
+)
+
+
+class FileWriter(object):
+    """Writes `Summary` protocol buffers to event files.
+    The `FileWriter` class provides a mechanism to create an event file in a
+    given directory and add summaries and events to it. The class updates the
+    file contents asynchronously. This allows a training program to call methods
+    to add data to the file directly from the training loop, without slowing down
+    training.
+    """
+
+    def __init__(self,
+                 logdir,
+                 graph=None,
+                 max_queue=10,
+                 flush_secs=120,
+                 filename_suffix='',
+                 graph_def=None):
+        """Creates a `FileWriter` and an event file.
+        On construction the summary writer creates a new event file in `logdir`.
+        The other arguments to the constructor control the asynchronous writes to
+        the event file:
+        *  `flush_secs`: How often, in seconds, to flush the added summaries
+           and events to disk.
+        *  `max_queue`: Maximum number of summaries or events pending to be
+           written to disk before one of the 'add' calls block.
+        Args:
+          logdir: A string. Directory where event file will be written.
+          graph: A `Graph` object, such as `sess.graph`.
+          max_queue: Integer. Size of the queue for pending events and summaries.
+          flush_secs: Number. How often, in seconds, to flush the
+            pending events and summaries to disk.
+          graph_def: DEPRECATED: Use the `graph` argument instead.
+        """
+        logdir = str(logdir)
+        self.event_writer = EventFileWriter(
+            logdir, max_queue, flush_secs, filename_suffix)
+
+    def get_logdir(self):
+        """Returns the directory where event file will be written."""
+        return self.event_writer.get_logdir()
+
+    def add_event(self, event, step=None, walltime=None):
+        """Adds an event to the event file.
+        Args:
+          event: An `Event` protocol buffer.
+        """
+        event.wall_time = time.time() if walltime is None else walltime
+        if step is not None:
+            event.step = int(step)
```

Comment:
This explicit cast doesn't feel like it should be necessary. If it is necessary, there should be a comment explaining why.